Helper/Config.php

```diff
@@ -163,42 +161,79 @@ public function getEnvironmentVariable(string $envVarName): ?string
      *
      * @return void
      */
-    public function setIsMerchantActive(bool $value): void
+    public function setIsMerchantActive(?int $storeId, bool $isActive): void
     {
-        $this->scopeWriter->save(self::IS_MERCHANT_ACTIVE, $value);
-        $this->flushCaches();
+        $token = $this->getAccessToken($storeId);
+        if ($token === null) { // no need to set isActive when no token exists
+            return;
+        }
+        $metadata = new ProtectMetadata($token, $isActive);
+        $this->setStoreMetadata($storeId, $metadata);
     }
 
     /**
      * Returns if the merchant is Active for NS8
      *
      * @return bool
      */
-    public function isMerchantActive(): bool
+    public function isMerchantActive(?int $storeId): bool
     {
-        return (bool) $this->scopeConfig->getValue(self::IS_MERCHANT_ACTIVE);
+        $metadata = $this->getStoreMetadata($storeId);
+        return $metadata !== null ? $metadata->isActive : false;
     }
 
     /**
-     * Retrieve an access token.
-     *
-     * @return string|null The NS8 Protect Access Token.
+     * Retrieve an access token for a given store.
+     * @param string $storeId ID of the store to retrieve token for
+     * @return string|null The NS8 Protect Access Token for the provided store.
+     */
+    public function getAccessToken(?int $storeId): ?string
+    {
+        $metadata = $this->getStoreMetadata($storeId);
+        return $metadata !== null ? $metadata->token : null;
+    }
+
+    public function setAccessToken(?int $storeId, string $token): void
+    {
+        $metadata = new ProtectMetadata($token, $this->isMerchantActive($storeId));
+        $this->setStoreMetadata($storeId, $metadata);
+    }
+
+    /**
+     * @return ProtectMetadata
+     */
+    public function getStoreMetadata(?int $storeId): ?\stdClass
+    {
+        $metadatas = $this->getStoreMetadatas();
+        $storeId = $storeId !== null ? $storeId : self::EMPTY_STORE_ID;
+        return isset($metadatas[$storeId]) ? $metadatas[$storeId] : null;
+    }
+
+    /**
+     * Get all access tokens, for all stores.
+     * Structured as { [storeId: string]: ProtectMetadata };
+     * @return ProtectMetadata[]
      */
-    public function getAccessToken(): ?string
+    public function getStoreMetadatas(): array
     {
-        return $this->encryptor->decrypt($this->scopeConfig->getValue(self::ACCESS_TOKEN_CONFIG_KEY));
+        $rawTokensJson = $this->encryptor->decrypt($this->scopeConfig->getValue(self::ACCESS_TOKEN_CONFIG_KEY));
+        $accessTokens = (array)json_decode($rawTokensJson);
+        return $accessTokens === null ? [] : $accessTokens;
```

Comment:
It may save a comparison here if we just do an implicit cast `return (array) $accesstoken`. If it's null, an empty array will be returned, if access Tokens is already an array then we're good to proceed.